src/App.php

```diff
@@ -1502,4 +1505,26 @@ public function redirect($toUrl)
 			$this->internalRedirect($toUrl);
 		}
 	}
+
+	/**
+	 * @brief Static declaration of Friendica routes.
+	 *
+	 * Supports:
+	 * - Route groups
+	 * - Variable parts
+	 * Disregards:
+	 * - HTTP method other than GET
+	 * - Named parameters
+	 *
+	 * Handler must be the name of a class extending Friendica\BaseModule.
+	 *
+	 * @param RouteCollector $routeCollector
+	 * @throws InternalServerErrorException
+	 */
```

Comment:
What about moving this method inside `Router`? I think this would be semantically more correct.

Or alternative to a `RouterFactory` which loads the routes into the `Router` (so we could define alternative routes f.e.)

Reply:
The problem is the call to `Hook::callAll` which means it needs to be bound to a running App.

Reply:
see above. We can call the Router as a parameter for `runFrontend()`, where we do have anything we need for the `Hook` call :-)

Reply:
I was about to object that the `App\Router` could be reused for something else, and then I noticed the namespace. Routes are defined over there now.